libkbfs/folder_branch_ops.go

```diff
@@ -4349,8 +4350,12 @@ func (fbo *folderBranchOps) getUnlinkPathBeforeUpdatingPointers(
 
 	// If the first op in this MD update is a resolutionOp, we need to
 	// inspect it to look for the *real* original pointer for this
-	// node.
-	if resOp, ok := md.data.Changes.Ops[0].(*resolutionOp); ok {
+	// node.  Though only do that if the op we're processing is
+	// actually a part of this MD object; if it's the latest cached
+	// dirOp, then the resOp we're looking at belongs to a previous
+	// revision.
+	if resOp, ok := md.data.Changes.Ops[0].(*resolutionOp); ok &&
+		(len(fbo.dirOps) == 0 || op != fbo.dirOps[len(fbo.dirOps)-1].dirOp) {
```

Comment:
Hmm why do we only need to compare with the last one but not  iterating through them all?

Reply:
Because we always call `notifyOneOp` under a lock right after appending to that list (under the same lock).  So it will always either be the last one, or not there at all.

Reply:
I see; thanks!